### ISSUE TITLE
Support jsconfig.json

### ISSUE THREAD

**Issue by aleclarson** [aleclarson/vite-tsconfig-paths#22]

**Comment by beforan** [comment]
For now, you can use a `tsconfig.json` in your js only project.

all `jsconfig,json` really does (for vs code, at least) is implicitly set `allowJs: true` and disable errors relating to actual typescript compilation.

I've found the following `tsconfig.json` to work fine to enable this plugin in a js only project:

```jsonc
{
  "compilerOptions": {
    // this is implicitly set in a `jsconfig.json` so we have to do it manually here
    "allowJs": true,

    "outDir": "dist", // this essentially disables the compilation errors by telling vs code where files would be put if it were to compile (which it won't because this isn't a real typescript project)

    // Do these as normal to your liking for your project
    "baseUrl": "src",
    "paths": {
      "*": ["src/*"]
    }
  },

  // only treat files in here as js source (thus ignoring `node_modules` and your fake output directory above)
  "include": ["src"]
}
```

**Comment by aleclarson** [comment]
Released in v3.5.0

**Comment by huyhoang160593** [comment]
How does this work ? My project use this and it not working as intended :(

**Comment by MichaelDeBoey** [comment]
@aleclarson I currently don't have the bandwidth to look into that unfortunately

**Comment by ibqn** [comment]
It is pity that there is a regression on that. @huyhoang160593 it was working on windows 11 as well for me at least in version 3.2.4 in this exemplary project https://github.com/ibqn/beni-15-puzzle

**Comment by huyhoang160593** [comment]
> It is pity that there is a regression on that. @huyhoang160593 it was working on windows 11 as well for me at least in version 3.6.0 in this exemplary project https://github.com/ibqn/beni-15-puzzle 
> 
> ```package.json
> "vite-tsconfig-paths": "^3.6.0"
> ```

It seem like in your `jsconfig.json` there is no "path" been defined? This plugin is basically doing nothing with your project for now

**Comment by ibqn** [comment]
@huyhoang160593 I use it for global importing i.e. for example

```shell
import { App } from 'app'
import 'index.css'
```

**Comment by diego-hourly** [comment]
Hello hello! Is there a chance to get the `jsconfig.json` support back? Maybe using `cosmiconfig` is possible? For now I got to convert the file to `tsconfig.json` and add some attributes. (I'm migrating from react-scripts to vite)

```js
```

**Comment by aleclarson** [comment]
It's not simply a version bump. Since we use `tsconfck` for file discovery and it doesn't support searching for both `tsconfig.json` and `jsconfig.json` in the same monorepo, we'll have to settle for either double-crawling or a new option that lets you (the plugin user) override the default `configName` with `"jsconfig.json"` in your Vite config (which means, `tsconfig.json` files wouldn't be used). At first glance, the double-crawling approach seems the best DX with a hopefully negligible performance hit.

**Comment by dominikg** [comment]
tsconfck 3 with its new async cache is a lot faster ([60-100 times faster!](https://github.com/dominikg/tsconfck/blob/main/docs/benchmark.md)) than tsconfck 2. So even with a double crawl it would be an improvement. But either way in regular projects users won't notice, there are just not enough files/directories for it to make a difference.

The real question is what you would do with the result of a double crawl. Typescript itself  uses tsconfig.json by default and you can tell it about jsconfig.json with a cli flag. To my knowledge, its not defined how precedence is handled in a scenario where there are mixed jsconfig.json and tsconfig.json files, so the easy solution here is to follow typescripts lead and macke the config name configurable.
If you allow both simultaneously, it would be your responsibility to check which one is closer to the input file, tsconfck does not offer a hybrid mode (and has no plans to, unless typescript itself made that move)

**Comment by aleclarson** [comment]
Support for `jsconfig.json` released in v4.3.0

**Comment by diego-hourly** [comment]
Thank you! @aleclarson
